application-load-balancer/tests/unit.tftest.hcl

```diff
@@ -271,17 +295,291 @@ run "domain_length_validation_tests" {
+    error_message = "Invalid ENVIRONMENT environment variable for aws_lambda_function.origin-secret-rotate-function"
+  }
+
+  assert {
+    condition     = aws_lambda_function.origin-secret-rotate-function.environment[0].variables.ROLEARN == "arn:aws:iam::${var.dns_account_id}:role/dbt_platform_cloudfront_token_rotation"
+    error_message = "Invalid ROLEARN environment variable for aws_lambda_function.origin-secret-rotate-function"
+  }
+
+  assert {
+    condition     = aws_lambda_function.origin-secret-rotate-function.environment[0].variables.AWS_ACCOUNT == data.aws_caller_identity.current.account_id
+    error_message = "Invalid AWS_ACCOUNT environment variable for aws_lambda_function.origin-secret-rotate-function"
+  }
+
+  assert {
+    condition     = aws_lambda_function.origin-secret-rotate-function.environment[0].variables.SLACK_TOKEN == data.aws_ssm_parameter.slack_token.value
+    error_message = "Invalid SLACK_TOKEN environment variable for aws_lambda_function.origin-secret-rotate-function"
+  }
+
+  assert {
+    condition     = aws_lambda_function.origin-secret-rotate-function.environment[0].variables.SLACK_CHANNEL == local.config_with_defaults.slack_alert_channel_alb_secret_rotation
+    error_message = "Invalid SLACK_CHANNEL environment variable for aws_lambda_function.origin-secret-rotate-function"
+  }
+
+  assert {
+    condition     = aws_lambda_permission.rotate-function-invoke-permission.statement_id == "AllowSecretsManagerInvocation"
+    error_message = "Invalid statement_id for aws_lambda_permission.rotate-function-invoke-permission"
+  }
+
+  assert {
+    condition     = aws_lambda_permission.rotate-function-invoke-permission.action == "lambda:InvokeFunction"
+    error_message = "Invalid action for aws_lambda_permission.rotate-function-invoke-permission"
+  }
+
+  assert {
+    condition     = aws_lambda_permission.rotate-function-invoke-permission.function_name == aws_lambda_function.origin-secret-rotate-function.function_name
+    error_message = "Invalid function_name for aws_lambda_permission.rotate-function-invoke-permission"
+  }
+
+  assert {
+    condition     = aws_lambda_permission.rotate-function-invoke-permission.principal == "secretsmanager.amazonaws.com"
+    error_message = "Invalid principal for aws_lambda_permission.rotate-function-invoke-permission"
+  }
+
+
+  assert {
+    condition     = aws_iam_role.origin-secret-rotate-execution-role.name == "${var.application}-${var.environment}-origin-secret-rotate-role"
+    error_message = "Invalid name for aws_iam_role.origin-secret-rotate-execution-role"
+  }
+
+  assert {
+    condition     = aws_iam_role.origin-secret-rotate-execution-role.assume_role_policy != null
+    error_message = "Invalid assume_role_policy for aws_iam_role.origin-secret-rotate-execution-role"
+  }
+
+  # Cannot assert against the arn in a plan. Requires an apply to evaluate.
+  # assert {
```

Comment:
Can get rid of the commented assert{} blocks in these - just leave the description.